src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -62,6 +63,21 @@ class GitVersionPlugin implements Plugin<Project> {
         }
     }
 
+    @Memoized
+    private String gitBranchName(Project project) {
+        Git git = gitRepo(project)
+        try {
+            def ref = git.repository.getRef(git.repository.branch)
+            if (ref == null) {
+                return null
+            } else {
+                return ref.getName().substring(Constants.R_HEADS.length())
+            }
+        } catch (Throwable t) {
+            return UNSPECIFIED_VERSION
```

Comment:
If https://github.com/palantir/gradle-git-version/pull/38 merges then this behaviour will need to change obviously.